### ISSUE TITLE
🐛 Fix alt text tags

### ISSUE THREAD

**Issue by ShanaLMoore** [notch8/utk-hyku#471]
related to: 
- https://github.com/scientist-softserv/utk-hyku/issues/43
- https://github.com/samvera/hyku/pull/1813/files
- https://github.com/scientist-softserv/utk-hyku/issues/393

# Summary

Per the accessibility audit, alt text needs to be appropriately used and applied. We discovered a few issues after using a screen reading tool (mac VoiceOver. [Extension](https://chrome.google.com/webstore/detail/alt-text-tester/koldhcllpbdfcdpfpbldbicbgddglodk)): 

- [ ] Logo Image: Update to use an alt attribute. 

![Image](https://github.com/scientist-softserv/utk-hyku/assets/10081604/4d226966-d407-4601-b809-517523919c37)

- [ ] Banner Image: 
  - N/A. Screen readers typically do not read the alt text of background images because they are considered decorative and not essential for conveying information. The alt attribute is specifically intended for images inserted within the HTML content. It may be misleading to allow users to set it in the dashboard/appearance section. It currently sets the title attribute. 
- [x]  Directory Image: TBD 
  - ~~Not sure where or if it's being used~~
  - This feature hasn't been contributed back to hyku yet. I will make an issue against hyku. It's being used in Pals

![Image](https://github.com/scientist-softserv/utk-hyku/assets/10081604/8d3f33d2-a879-4371-b189-f0e14323dcd1)

- [ ]  Default Work Image:   
  a. Thumbnail: Missing alt text

![Image](https://github.com/scientist-softserv/utk-hyku/assets/10081604/d1ff586b-abfa-44f4-94fb-585de589c086)

  b. Default Image: Not displaying after being set

![Image](https://github.com/scientist-softserv/utk-hyku/assets/10081604/fe412da8-6818-4407-82ef-3f7b618e2aa7)

- [ ] Default Collection Image: 
  a. Thumbnail: alt tag should be dynamic

![Image](https://github.com/scientist-softserv/utk-hyku/assets/10081604/cf07dd42-974a-412f-8dc6-afe705cc2359)

  b. Default Image: alt tag is empty and should be dynamic

![Image](https://github.com/scientist-softserv/utk-hyku/assets/10081604/c977a6a6-074b-4468-8d47-a3912d4fec20)

**Note: Default Images are used when the user creates a work without a fileset, or a collection without an image.** 

# Acceptance Criteria

- [ ] If an alt text is set, it should be used. Inspect elements to confirm the alt tag is present and set to the appropriate values. 

# Screenshots or Video

# Testing Instructions

It may be best to consult Diem who originally performed the accessibility audit and took note of failures. Retest this using the rubric she had. 

ref: [Report.for.https___test.hykucommons.org_.pdf](https://github.com/scientist-softserv/utk-hyku/files/11287664/Report.for.https___test.hykucommons.org_.pdf)

Otherwise...

Set all of the options in the appearance setting.
Confirm the alt tags are in the inspect elements in all of the places. Reference the bullet points in the summary section for guidance. 

# Notes

- [ ] #474

Issue filed against hyku for missing directory image partial: https://github.com/samvera/hyku/issues/1967

TODO:
- [x] AllinsonFlex needs an alt tag added to https://github.com/samvera-labs/allinson_flex/blob/main/app/views/hyrax/my/works/_list_works.html.erb#L12C10-L12C10 . 
  - [x] Compare it to hyrax that included the alt tag, however it needs to be dynamic to pull in the alt text that we set it to in hyku. default_work_image_text
- [x] IIIFPrint needs an update to representative_media. It currently references a default image when it should pull in the image we set it to in hyku. Default_work_image. 
  - [x] The alt text should be updated to be dynamic. pull in what is set in hyku. 
- [ ] ~~Hyku needs a partial override/copy of point number 2 to override hyrax proper which just uses the default image.~~ 
  - this will get updated when hyku updates iiif_print
- [x] UTK media_display alt tag needs to dynamically pull in the settings from hyky. default_collection_image_text

**Comment by DiemBTran** [comment]
**Needs further review:**

tested on:
- Settings Appearance tab: https://qa.utk-hyku-staging.notch8.cloud/admin/appearance
- logo from dashboard test: https://qa.utk-hyku-staging.notch8.cloud/dashboard?locale=en
- logo and banner from homepage: https://qa.utk-hyku-staging.notch8.cloud/?locale=en
- default work icon: https://qa.utk-hyku-staging.notch8.cloud/dashboard/my/works?locale=en
- default collection icon: https://qa.utk-hyku-staging.notch8.cloud/dashboard/my/collections?locale=en

testing instructions:
- [ ] logo alt text is correct
  - from the home page, this is correct
  - from the dashboard, this is incorrect. should be `alt` attribute, not `alt_text` attribute 
- [x] banner alt text is correct
- [ ] directory alt text is correct
  - cannot be tested. N/A to UTK bc they don't have a directory splash page 
- [ ] default collection image alt text is correct
  - alt text doesn't save properly
- [ ] default work image alt text is correct
  - alt text doesn't save properly

<details><summary>click to expand</summary>

![Hyku 2023-07-26 at 9 27 57 AM](https://github.com/scientist-softserv/utk-hyku/assets/29311858/0d26ca1d-6950-49f2-b84a-d632e0113ed9)
![Show Appearance Hyku 2023-07-26 at 9 24 56 AM](https://github.com/scientist-softserv/utk-hyku/assets/29311858/89e0e994-9d8e-4eb2-9d26-ca9f18da213f)
![Hyku 2023-07-26 at 9 28 46 AM](https://github.com/scientist-softserv/utk-hyku/assets/29311858/08900398-878e-4249-b667-9bfed4efbea1)
![Show Appearance Hyku 2023-07-26 at 9 29 42 AM](https://github.com/scientist-softserv/utk-hyku/assets/29311858/ec05ab99-8595-40ea-915c-20aea585a6fc)
![Collections 2023-07-26 at 9 31 41 AM](https://github.com/scientist-softserv/utk-hyku/assets/29311858/2c8d1ac5-2da8-4345-9a69-53bb04adcc51)
![Works 2023-07-26 at 9 30 42 AM](https://github.com/scientist-softserv/utk-hyku/assets/29311858/11a643d2-dec7-4b89-ad0d-f453c6a31961)

</details>

**Comment by ShanaLMoore** [comment]
- [x] logo alt text is correct
  - from the home page, this is correct
  - [x] from the dashboard, this is incorrect. should be `alt` attribute, not `alt_text` attribute 

<img width="712" alt="image" src="https://github.com/scientist-softserv/utk-hyku/assets/10081604/f18b86bf-32c8-4185-b664-2f33a2ea718a">

- [x] banner alt text is correct

- [ ] collection thumbnail has the incorrect alt value

- [x] directory alt text is correct
  - cannot be tested. N/A to UTK bc they don't have a directory splash page (that is correct ✅ )

**For the following two comments, if you only edit input field for alt text and click save, the form doesn't actually save. You. must re upload an image as well.** 

- [ ] default collection thumbnail image alt text is correct
  -  alt text doesn't save properly
- [ ] default work thumbnail image alt text is correct
  - alt text doesn't save properly

**Comment by ShanaLMoore** [comment]
cc @kirkkwang fyi rework branch started: i43-alt-text-tag-followup

TODO: 

track down and fix collection/work thumbnail alt texts per Diem's feedback ^^

**Comment by jillpe** [comment]
- [x] logo alt text is correct
<details><summary>Screenshots</summary>
<img width="1268" alt="Screen Shot 2023-08-01 at 2 39 16 PM" src="https://github.com/scientist-softserv/utk-hyku/assets/84697174/33fcf1d6-d9ef-49b9-8f76-87593dde9816">



![Image](https://github.com/scientist-softserv/utk-hyku/assets/84697174/8e4dddf4-fa93-4c14-b408-fb53d24f7848)



</details

- [x]  banner alt text is correct
<details><summary>Screenshots</summary>


![Image](https://github.com/scientist-softserv/utk-hyku/assets/84697174/330acd09-796a-418e-ba6e-d140e3d180c3)



</details>

Directory alt text is N/A for UTK because they do not use this feature.


- [ ] default collection image alt text is correct
<details><summary>Screenshots</summary>
See Shana's comment below for more details


![Image](https://github.com/scientist-softserv/utk-hyku/assets/84697174/af32ba87-5efb-440c-880b-1cbf8a35e43e)


</details>

- [x] default work image alt text is correct
<details><summary>Screenshots</summary>


![Image](https://github.com/scientist-softserv/utk-hyku/assets/84697174/5e5135f9-e7ce-425e-887a-3537a2fa8699)




</details>

**Comment by ShanaLMoore** [comment]
# QA Results: ❌ 

After meeting with Jill's assessment, this ticket ...
<details>
<summary> needs rework because when i set the alt text for the collection splash page banner image, it set the title element instead of an alt tag. Images should have an alt tag for accessibility reasons.  </summary>

![image](https://github.com/scientist-softserv/utk-hyku/assets/10081604/5c952ce9-d9a5-4b48-af88-98320e35fd3b)

</details>

<details> 

<summary> The user must set this by editing a collection from the admin dashboard's branding tab. They would upload an image and set the alt text there too. TODO: For rework, confirm that setting the logo also sets the alt element and not just the title of the image. </summary> 

![image](https://github.com/scientist-softserv/utk-hyku/assets/10081604/e049aa38-94bc-45a9-82ce-93d0bbeecf16)

</details>

**Comment by kirkkwang** [comment]
@ShanaLMoore Since this a `div` tag i don't expect it to have an alt text.  Perhaps the setting of it should reflect that it is setting the title?

**Comment by ShanaLMoore** [comment]
OHHH yea you're right! @kirkkwang  🙃 

But should it be a div tag? 🤔 I agree the setting label is misleading if it should be. 

What about the collection logo? 

If that's good too then this passes cc @jillpe

**Comment by ShanaLMoore** [comment]
> OHHH yea you're right! @kirkkwang 🙃
> 
> But should it be a div tag? 🤔 I agree the setting label is misleading if it should be.
> 
> What about the collection logo?
> 
> If that's good too then this passes cc @jillpe

We're good

the logo has an alt tag and an img element

![image](https://github.com/scientist-softserv/utk-hyku/assets/10081604/7e9c30d2-9267-498b-8da6-02118e245c5d)

**Comment by markpbaggett** [comment]
Looks good to me.  Bumping.
